gdp/dsda/dsda_functions.py

```diff
@@ -1278,44 +1430,68 @@ def solve_with_dsda(
     return m2_solved, route, obj_route
 
 
-def visualize_dsda(
+def visualize_dsda( # TODO: Modify the code, I think scatter should be upper than the arrows. I suggest to move sc above for loop.
```

Comment:
Comments should also be well written. Moreover, move the comments to within the function and not where it is defined.